src/test/java/com/example/dog/controller/DogControllerRestAssuredMockMvcTest.java

```diff
@@ -164,4 +171,19 @@ private static Dog getDogFromDeleteRequest(int id) {
     private static MockMvcResponse deleteDogAndReturn(int id) {
         return given().when().delete("/dog/{id}", id).thenReturn();
     }
+
+    /**
+     * Initializes GlobalControllerExceptionHandler advice using the StaticApplicationContext with the single bean.
+     * @return HandlerExceptionResolver instantiated based on the GlobalControllerExceptionHandler
+     */
+    private HandlerExceptionResolver initGlobalExceptionHandlerResolvers() {
```

Comment:
Ok, in the next commit you can try creating MockMvc from the existing Spring Context instead of creating a standaloneSetup(). Looks something like this:  `MockMvcBuilders.webAppContextSetup(applicationContext).build()`

This way you won't need to duplicate production set up - you will _use_ production set up.